Riot/Modules/SetPinCode/SetupBiometrics/SetupBiometricsViewModel.swift

```diff
@@ -67,32 +71,42 @@ final class SetupBiometricsViewModel: SetupBiometricsViewModelType {
         case .unlock:
             unlockWithBiometrics()
         case .cantUnlockedAlertResetAction:
-            coordinatorDelegate?.setupBiometricsViewModelDidCompleteWithReset(self)
+            coordinatorDelegate?.setupBiometricsViewModelDidCompleteWithReset(self, dueToTooManyErrors: false)
         }
     }
     
     // MARK: - Private
     
     private func enableDisableBiometrics() {
-        LAContext().evaluatePolicy(.deviceOwnerAuthentication, localizedReason: VectorL10n.biometricsUsageReason) { (success, error) in
-            if success {
-                //  complete after a little delay
-                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-                    self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
-                }
+        biometricsAuthenticationPresenter.present(with: VectorL10n.biometricsUsageReason) { (response) in
+            switch response {
+            case .success:
+                self.pinCodePreferences.canUseBiometricsToUnlock = nil
+                self.pinCodePreferences.resetCounters()
+                self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
+            case .failure:
+                break
             }
         }
     }
     
     private func unlockWithBiometrics() {
-        LAContext().evaluatePolicy(.deviceOwnerAuthentication, localizedReason: VectorL10n.biometricsUsageReason) { (success, error) in
-            if success {
-                //  complete after a little delay
-                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-                    self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
+        biometricsAuthenticationPresenter.present(with: VectorL10n.biometricsUsageReason) { (response) in
+            switch response {
+            case .success:
+                self.pinCodePreferences.canUseBiometricsToUnlock = nil
+                self.pinCodePreferences.resetCounters()
+                self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
+            case .failure(let error):
+                guard let error = error as NSError? else {
+                    return
                 }
```

Comment:
Error is bridgeable to NSError (like String and NSString) you can do:
```suggestion
                let nsError = error as NSError
```

Reply:
Yes, but it's intentional to simplify two lines into one